internal/cookie/cookie.go

```diff
@@ -0,0 +1,15 @@
+package cookie
+
+import (
+	"net/http"
+
+	"github.com/cockroachdb/errors"
+)
+
+func AnonymousUID(r *http.Request) (string, error) {
+	cookie, err := r.Cookie("sourcegraphAnonymousUid")
+	if err != nil {
+		return "", errors.Wrap(err, "getting cookie value")
+	}
+	return cookie.Value, nil
+}
```

Comment:
Why even return an error if it's ignored in the majority of cases?

```suggestion
func AnonymousUID(r *http.Request) string {
	cookie, err := r.Cookie("sourcegraphAnonymousUid")
	if err != nil {
		return ""
	}
	return cookie.Value
}
```

Or return a `(string, bool)`?

Reply:
Because `r.Cookie` *could* return an error I don't want to make the decision to ignore it for all consumers.

Reply:
Yeah, but _all_ call sites that you updated in this branch ignore the error. Either by assigning to `_` or by checking `if err == nil` and only then using the value, but none of them uses the error as a signal to abort. All they care about is "can I use this value?"

Reply:
But the only reason `r.Cookie` could return an error for is if the named cookie doesn't exist.

> Cookie returns the named cookie provided in the request or ErrNoCookie if not found. If multiple cookies match the given name, only one cookie will be returned.

So if the empty string can be used to signal that, that's a simpler interface to work with. I think we should change it.

Reply:
I think just returning a string would be a little misleading and would force the consumer to do the check. So,, I think I'll go with what @mrnugget suggested since in this case all we are about was whether the cookie exists or not.